Develop

bdilday opened the pull request:
This pull request applies the redo-time changes and provides sqlite support

mhoover committed:
Make changes to record time as HH:MM and use throughout.

bdilday committed:
initial requirements file

bdilday committed:
Merge remote-tracking branch 'mhoover/redo-time' into develop

mhoover committed:
Change default option on creating database.

bdilday committed:
WIP

bdilday committed:
provide sqlite as an option

bdilday committed:
add sqlite option

bdilday committed:
add sqlite option

bdilday committed:
read all fields defined in config file into dictionary

bdilday committed:
add sqlite support

bdilday committed:
fixed merge conflicts

bdilday committed:
applied redo-time updates

bdilday committed:
Merge pull request #1 from bdilday/sqlite-support

Sqlite support

bdilday committed:
updated time format

mhoover commented on create.py:
```diff
@@ -3,43 +3,79 @@
                         'create.')
     parser.add_argument('--host', required=False, help='Database host; will default to '
                         'config settings.')
+    parser.add_argument('--dbengine', required=False, help='Database engine; will default to '
```
Provide specific options, i.e., `parser.add_argument('-e', '--dbengine', required=False, choices=['mysql', 'sqlite'], help='Database engine; will default to config settings.')` that are supported.

mhoover commented on add.py:
```diff
@@ -2,21 +2,24 @@
+                         password='{}'.format(PASSWORD), db=args_dict['db'],
+                         charset='utf8', autocommit=True)
+    elif args_dict['dbengine'] == 'sqlite':
+        db = sqlite3.connect('{}.db'.format(args_dict['db']), isolation_level=None)
```
Let's close this out with:
```
else:
     sys.exit('You\'re trying to use a DB engine that's not currently supported. Try again, specifying `mysql` or `sqlite`.')
```

using line breaks if need be to keep the line under 100 characters long.

bdilday replied:
sure. any reason not to use a `ValueError` exception?

mhoover replied:
`ValueError` is cool; I used `sys.exit()` at another point, so that's the only reason for the suggestion. I'm fine with a `ValueError`.

mhoover commented on analyze.py:
```diff
@@ -43,15 +45,19 @@ def run(args_dict):
+        db = mdb.connect(user='{}'.format(USERNAME), host=args_dict['host'],
+                         password='{}'.format(PASSWORD), db=args_dict['db'], charset='utf8')
+    elif args_dict['dbengine'] == 'sqlite':
+        db = sqlite3.connect('{}.db'.format(args_dict['db']), isolation_level=None)
```
Same thing here -- let's close this out with an `else:` statement that exits if someone tries to use an invalid DB engine.

mhoover commented on analyze.py:
```diff
@@ -43,15 +45,19 @@ def run(args_dict):
     df = df[(df.date>=args_dict['date'][0]) & (df.date<=args_dict['date'][1])]
-
+    df.start.apply(lambda x: x/np.timedelta64(1, 'm'))
+    df.end.apply(lambda x: x/np.timedelta64(1, 'm'))
```
These two apply statements -- please delete. These will be addressed shortly in a PR.

mhoover commented on create.py:
```diff
@@ -3,43 +3,79 @@
+    '''.format(table=args_dict['table']))]
+
+    return sql_commands
+
```
Can we make these one function like `create_table_command`, passing the `args_dict['dbengine']` as a parameter to determine which set of commands to follow?

bdilday replied:
I'd suggest handling that in a separate pull request, one that also defines a function to make the db connection based on the value of `dbengine`. Otherwise there will still be the need for the `if else` to decide to use `mdb.connect` or `sqlite3.connect` before calling `create_table_command`.

mhoover replied:
Fair point. Okay, then keep these as is for now and we'll address it later.

mhoover commented on create.py:
```diff
@@ -3,43 +3,79 @@
+                         ''''''.format(table=args_dict['table'])))
+
+    for sql in sql_commands:
+        db.cursor().execute(sql)
```
Have these changes been tested against both mysql and sqlite?

bdilday replied:
no just sqlite.

mhoover replied:
See my comment above -- this needs to be tested before merging. If you don't have mysql installed, I can test.

mhoover reviewed (commented):
Nice work. Just a couple of things to touch up.

mhoover commented:
@bdilday Let's get these changes made, this PR merged, then I can rebase #4 off of the merged work. If you could review #4 as well, that'd be great.

mhoover commented on create.py:
```diff
@@ -3,43 +3,79 @@
+    ;
+    '''.format(db=args_dict['db'], table=args_dict['table']))]
+
+    sql_commands += ['create database if not exists {}'.format(args_dict['db'])]
```
As I was looking at this this is going to append the list with the `create database...` *after* the `use database...` call, which should result in an error. I think this goes to testing -- I can test this in mysql if you don't have mysql installed. LMK.

bdilday replied:
Yes it looks out of order. I'm not set up to test mysql currently.

bdilday committed:
resolve issues from PR code review

mhoover committed:
Fix up DB creation for mysql.

This commit touches up work to ensure changes in creating databases
and tables are workable for both mysql and sqlite. These changes
have been tested against both database types.

mhoover committed:
Update README.

mhoover commented:
Alright, this is good to go. 👍 

Merging to rebase other work.
